src/models/stores/sortable-documents.ts

```diff
@@ -0,0 +1,196 @@
+      return numA - numB;
+    });
+    return sortedSectionLabels.map(sectionLabel => documentMap.get(sectionLabel));
+  }
+
+  //******************************************* Sort By Name **************************************
+  get sortByName(): SortedDocument[]{
+    const documentMap = new Map();
+    this.filteredDocsByType.forEach((doc) => {
+      const user = this.class.getUserById(doc.uid);
+      const sectionLabel = (user && user.type === "student") ? `${user.lastName}, ${user.firstName}` : "Teacher";
+      if (!documentMap.has(sectionLabel)) {
+        documentMap.set(sectionLabel, {
+          sectionLabel,
+          documents: []
+        });
+      }
+      documentMap.get(sectionLabel).documents.push(doc);
+    });
+
+    const sortedSectionLabels = Array.from(documentMap.keys()).sort((a, b) => {
+      const parseName = (name: any) => {
+        const [lastName, firstName] = name.split(", ").map((part: any) => part.trim());
+        return { firstName, lastName };
+      };
+      const aParsed = parseName(a);
+      const bParsed = parseName(b);
+
+      // Compare by last name, then by first name if last names are equal
+      const lastNameCompare = aParsed.lastName.localeCompare(bParsed.lastName);
+      if (lastNameCompare !== 0) {
+        return lastNameCompare;
+      }
+      return aParsed.firstName.localeCompare(bParsed.firstName);
+    });
+    return sortedSectionLabels.map(sectionLabel => documentMap.get(sectionLabel));
+  }
+
+  //*************************************** Sort By Strategy **************************************
+
+  get sortByStrategy(): SortedDocument[]{
+    const commentTags = this.commentTags;
+    const tagsWithDocs: Record<string, TagWithDocs> = {};
+    if (commentTags) {
+      for (const key of Object.keys(commentTags)) {
+        tagsWithDocs[key] = {
+          tagKey: key,
+          tagValue: commentTags[key],
+          docKeysFoundWithTag: []
+        };
+      }
+      tagsWithDocs[""] = { //this accounts for when user commented with tagPrompt (no tag selected)
+        tagKey: "",
+        tagValue: "Not Tagged",
+        docKeysFoundWithTag: []
+      };
+    }
+
+    this.tempTagDocumentMap.forEach((docKeysSet, tag) => {
+      const docKeysArray = Array.from(docKeysSet); // Convert the Set to an array
+      if (tagsWithDocs[tag]) {
+        tagsWithDocs[tag].docKeysFoundWithTag = docKeysArray;
+      }
+    });
+
+    const sortedDocsArr: SortedDocument[] = [];
+
+    Object.entries(tagsWithDocs).forEach((tagKeyAndValObj) => {
+      const tagWithDocs = tagKeyAndValObj[1] as TagWithDocs;
+      const sectionLabel = tagWithDocs.tagValue;
+      const docKeys = tagWithDocs.docKeysFoundWithTag;
+      const documents = this.documents.all.filter(doc => docKeys.includes(doc.key));
+      sortedDocsArr.push({
+        sectionLabel,
+        documents
+      });
+    });
+
+    return sortedDocsArr;
+  }
+
+  //Ideas with Scott:
```

Comment:
Leaving this here for future ideas on how to speed up our firestore query.

Reply:
And just as reference - here is the old "original" code that asked for all the documents.
The whole function`updateTagDocumentMap` is pasted below. 

  updateTagDocumentMap () {
    //------------------ old --------------------

    const db = this.db.firestore;
    const filteredDocs = this.filteredDocsByType;
    const documentIDs= filteredDocs.map(doc => {
     return `uid:${doc.uid}_${doc.key}`;
    });

    const unsubscribeFromDocs = db.collection("documents").onSnapshot(docsSnapshot => {
      console.log("docSnapshot!", docsSnapshot);

      docsSnapshot.forEach(doc => {
        const docData = doc.data();
        // console.log("doc path:", doc.ref.path);
        const docKey = docData.key;
        const commentsRef = doc.ref.collection("comments"); //access sub collection
        commentsRef.get().then(commentsSnapshot => {
          // console.log("--------------------");
          // console.log(docKey);
          // console.log("\t🥩 commentsSnapshot:", commentsSnapshot);
          runInAction(() => {
            commentsSnapshot.forEach(commentDoc => {
              const commentData = commentDoc.data();
              if (commentData && commentData.tags) {
                // console.log("old - enters if");
                // console.log("commentData:", commentData);
                commentData.tags.forEach((tag: string) => {
                  let docKeysSet = this.tempTagDocumentMap.get(tag);
                  if (!docKeysSet) {
                    docKeysSet = new ObservableSet<string>();
                    this.tempTagDocumentMap.set(tag, docKeysSet);
                  }
                  console.log("doc path:", doc.ref.path);
                  console.log("\t🔪 docKey:", docKey);
                  console.log("\t🔪 tag:", tag);

                  docKeysSet.add(docKey); //only unique doc keys will be stored
                });
              }
            });
            // console.log("tempTagDocumentMap:", this.tempTagDocumentMap);

          });

        });
      });
      unsubscribeFromDocs();
    });

    //------end old----------------

  }